diag_manager/diag_util.F90

```diff
@@ -621,114 +629,113 @@ END FUNCTION get_index
+      lname  = TRIM(long_name)
     ELSE
-       lname  = ''
+      lname  = ''
     END IF
 
     IF ( PRESENT(units) ) THEN
-       lunits = TRIM(units)
+      lunits = TRIM(units)
     ELSE
-       lunits = ''
+      lunits = ''
     END IF
 
     WRITE (numaxis,'(i1)') SIZE(axes)
 
     IF (PRESENT(missing_value)) THEN
-       IF ( use_cmor ) THEN
-          WRITE (lmissval,*) CMOR_MISSING_VALUE
-       ELSE
-          SELECT TYPE (missing_value)
+      IF ( use_cmor ) THEN
+        WRITE (lmissval,*) CMOR_MISSING_VALUE
+      ELSE
+        SELECT TYPE (missing_value)
           TYPE IS (real(kind=r4_kind))
-             missing_value_use = missing_value
+            missing_value_use = missing_value
           TYPE IS (real(kind=r8_kind))
-             missing_value_use = real(missing_value)
+            missing_value_use = real(missing_value)
           CLASS DEFAULT
-             CALL error_mesg ('diag_util_mod::log_diag_field_info',&
-                  & 'The missing_value is not one of the supported types of real(kind=4) or real(kind=8)', FATAL)
-          END SELECT
-          WRITE (lmissval,*) missing_value_use
-       END IF
+            CALL mpp_error( FATAL,'diag_util_mod::log_diag_field_info: The missing_value is not one of the supported'//&
+                          & ' types of real(kind=4) or real(kind=8)')
+        END SELECT
+        WRITE (lmissval,*) missing_value_use
+      END IF
     ELSE
-       lmissval = ''
+      lmissval = ''
     ENDIF
 
     IF ( PRESENT(range) ) THEN
-       SELECT TYPE (range)
-       TYPE IS (real(kind=r4_kind))
+      SELECT TYPE (range)
+      TYPE IS (real(kind=r4_kind))
           range_use = range
-       TYPE IS (real(kind=r8_kind))
+      TYPE IS (real(kind=r8_kind))
           range_use = real(range)
-       CLASS DEFAULT
-          CALL error_mesg ('diag_util_mod::log_diag_field_info',&
-               & 'The range is not one of the supported types of real(kind=4) or real(kind=8)', FATAL)
-       END SELECT
-       WRITE (lmin,*) range_use(1)
-       WRITE (lmax,*) range_use(2)
+      CLASS DEFAULT
+          CALL mpp_error('diag_util_mod::log_diag_field_info',&
+              & 'The range is not one of the supported types of real(kind=4) or real(kind=8)', FATAL)
+      END SELECT
+      WRITE (lmin,*) range_use(1)
+      WRITE (lmax,*) range_use(2)
     ELSE
-       lmin = ''
-       lmax = ''
+      lmin = ''
+      lmax = ''
     END IF
 
     IF ( PRESENT(dynamic) ) THEN
-       IF (dynamic) THEN
+      IF (dynamic) THEN
           timeaxis = 'T'
-       ELSE
+      ELSE
           timeaxis = 'F'
-       END IF
+      END IF
     ELSE
-       timeaxis = ''
+      timeaxis = ''
     END IF
 
-    axes_list=''
-    DO i = 1, SIZE(axes)
-       CALL get_diag_axis_name(axes(i),axis_name)
-       IF ( TRIM(axes_list) /= '' ) axes_list = TRIM(axes_list)//','
-       axes_list = TRIM(axes_list)//TRIM(axis_name)
-    END DO
+    if ( .not. wrote_header ) then
```

Comment:
Is this thread safe?  This is a local saved variable, so won't it happen twice if there are 2 threads?